### PR TITLE
Change `grabclipboard()` to use PNG compression on macOS

### LINKED ISSUE #7214: PIL.ImageGrab.grabclipboard() is lossy on Mac
### Issue

On Mac, `PIL.ImageGrab.grabclipboard()` is lossy because it uses JPG:

```python
commands = [
            'set theFile to (open for access POSIX file "'
            + filepath
            + '" with write permission)',
            "try",
            "    write (the clipboard as JPEG picture) to theFile",
            "end try",
            "close access theFile",
        ]
```

This results in lossy acquisition, which can be highly undesirable.

One such cases is when dealing with screenshots. The noise induced by the JPG conversion on otherwise relatively flat image content eventually leads to much bigger PNG output (if such format is desired in the end).

This can be easily fixed with the following change:

```python
commands = [
            'set theFile to (open for access POSIX file "' + filepath + '" with write permission)',
            "try",
            "    write (the clipboard as «class PNGf») to theFile",
            "end try",
            "close access theFile",
        ]
```

Happy to submit a PR if there is willingness to apply this change.

### What are your OS, Python and Pillow versions?

* OS: macOS Ventura
* Python: 3.11
* Pillow: 9.5

### PR DESCRIPTION
**Pull request by abey79**
This PR changes `grabclipboard()` to use a PNG compression instead of a lossy JPG compression when reading images from he clipboard on macOS.

Fixes #7214

### CODE REVIEW

**Comment by radarhere** [comment]
A note for the future - our tradition is to not include the CHANGES.rst updates in PRs. If something else were to be merged in first, it would create a conflict.

I've created https://github.com/abey79/Pillow/pull/1 with a suggestion.

**Review by radarhere — APPROVED** [review]

**Comment by abey79** [comment]
Thanks for the fix, quite the overlook on my part.

As for CHANGES.rst, it's noted. Maybe `.github/CONTRIBUTING.md` might be reworded? I tried to comply to the following based on recent commits I found:

> - Include [release notes](https://github.com/python-pillow/Pillow/tree/main/docs/releasenotes) as needed or appropriate with your bug fixes, feature additions and tests.

File: src/PIL/ImageGrab.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def grabclipboard():
             + filepath
             + '" with write permission)',
             "try",
-            "    write (the clipboard as JPEG picture) to theFile",
+            "    write (the clipboard as «class PNGf») to theFile",
```
Comment (hugovk) [inline src/PIL/ImageGrab.py:105]: Please could you explain the `«class PNGf»` syntax?

Comment (abey79) [reply src/PIL/ImageGrab.py:105]: I'm not sure—I'm very far from an osascript expert. It seems that this class of image format doesn't have a proper name in the language, e.g. see: https://apple.stackexchange.com/a/375353/53910

I've tried using `PNG picture`, but it certainly doesn't work.

Comment (radarhere) [reply src/PIL/ImageGrab.py:105]: If you create a screenshot on the clipboard with Command-Control-Shift-3, and then run `clipboard info` in AppleScript, here is the result - a list that includes 'JPEG picture' and '«class PNGf»'.
Screenshot 2023-06-18 at 1 58 21 pm

**Comment by hugovk** [comment]
Thanks!

**Comment by radarhere** [comment]
> As for CHANGES.rst, it's noted. Maybe `.github/CONTRIBUTING.md` might be reworded? I tried to comply to the following based on recent commits I found:
> 
> > * Include [release notes](https://github.com/python-pillow/Pillow/tree/main/docs/releasenotes) as needed or appropriate with your bug fixes, feature additions and tests.

Ok, I've created PR https://github.com/python-pillow/Pillow/pull/7220

**Pull request merged**